app/Repositories/PostRepository.php

```diff
@@ -83,10 +83,35 @@ public function create(array $data, $signupId)
             'url' => $fileUrl,
             'caption' => $data['caption'],
             'status' => isset($data['status']) ? $data['status'] : 'pending',
-            'source' => $data['source'],
-            'remote_addr' => $data['remote_addr'],
+            'source' => isset($data['source']) ? $data['source'] : null,
+            'remote_addr' => isset($data['remote_addr']) ? $data['remote_addr'] : null,
+
         ]);
 
+        // If we are supporting quantity on posts and
+        // we recieved a quantity in the request.
+        // Then, store correct quantity on the Post.
+        if (config('features.v3QuantitySupport') && isset($data['quantity'])) {
+            $quantityDiff = $data['quantity'] - $signup->quantity;
+
+            // If the quantity difference is negative than we recieved an incremental submission
+            // and should just add that to the post.
+            if ($quantityDiff < 0) {
+                $quantityDiff = $data['quantity'];
+            } elseif ($quantityDiff === 0 && $signup->posts_count > 0) {
+                // If the quantity difference equals zero, and this is not the first post,
+                // then we can assume there is no difference in quantity and store it as 0 on the post.
+                $quantityDiff = 0;
+            }
+
+            $post->quantity = $quantityDiff;
+            $post->save();
```

Comment:
I don't think we need this since we call `$post->save()` below (so no need to hit the DB twice).